### PR TITLE
Support for HwSettle in Deployment Manager

### PR DESCRIPTION
**Pull request by sselvara1**

### CODE REVIEW

File: examples/standard/hw-settle/hw-settle.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: starlingx.windriver.com/v1
+kind: HostProfile
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: controller-profile
+  namespace: deployment
+spec:
+  hwSettle: 0
```
Comment (yjian118) [inline examples/standard/hw-settle/hw-settle.yaml:9]: Based on the example, what's the expected value of the hwSettle? "0" and "1"?

File: go.mod [hunk 2]
```diff
@@ -91,4 +91,4 @@ require (
 )
 
 // replace github.com/gophercloud/gophercloud => ./external/gophercloud
-replace github.com/gophercloud/gophercloud => github.com/Wind-River/gophercloud v0.0.0-20230119154844-fa8a1286cb56
+replace github.com/gophercloud/gophercloud => github.com/sselvara1/gophercloud v0.0.0-20230223101244-77c13c232f8c
```
Comment (yjian118) [inline go.mod:94]: Before merging the code, don't forget to change back the gophercloud commit to the commit in repo github.com/Wind-River/gophercloud

File: api/v1/hostprofile_types.go [hunk 3]
```diff
@@ -854,6 +854,10 @@ type ProfileBaseAttributes struct {
 	// AppArmor defines the security model on the host.
 	// +optional
 	AppArmor *string `json:"appArmor,omitempty"`
+
+	// HwSettle defines the wait time for SCSI devices to show up.
+	// +optional
```
Comment (ttakenak) [inline api/v1/hostprofile_types.go:859]: If this parameter only takes number, could you consider to add validation annotation? (You could refer "MaxCPUMhzConfigured" above)

File: examples/standard/hw-settle/hw-settle.yaml (hunk 1 above)
Comment (sselvara1) [reply examples/standard/hw-settle/hw-settle.yaml:9]: Based on the example, expected value of hwSettle is "0"

File: go.mod (hunk 2 above)
Comment (sselvara1) [reply go.mod:94]: sure

File: api/v1/hostprofile_types.go (hunk 3 above)
Comment (sselvara1) [reply api/v1/hostprofile_types.go:860]: yes, this parameter only takes number. added validation annotation.

File: config/crd/bases/starlingx.windriver.com_hostprofiles.yaml [hunk 4]
```diff
@@ -199,6 +199,10 @@ spec:
                 description: Console defines the installation output device.
                 pattern: ^(|tty[0-9]+|ttyS[0-9]+(,\d+([a-zA-Z0-9]+)?)?|ttyUSB[0-9]+(,\d+([a-zA-Z0-9]+))?|lp[0-9]+)$
                 type: string
+              hwSettle:
```
Comment (yjian118) [inline config/crd/bases/starlingx.windriver.com_hostprofiles.yaml:202]: Seem this actual CRD is not updated after adding the validation, need to re-generate the CRD.

**Comment by yjian118** [comment]
Apart from the CRDs, the code looks fine to me. Just checked with the dev prime of this feature,  the update of this argument is time sensitive, needs to have a re-test in a standard system, and the change in the kernel should also be checked/verified as the validation of this change.

File: config/crd/bases/starlingx.windriver.com_hostprofiles.yaml (hunk 4 above)
Comment (sselvara1) [reply config/crd/bases/starlingx.windriver.com_hostprofiles.yaml:202]: Re-generated CRD for update after adding the validation.

File: config/crd/bases/starlingx.windriver.com_hosts.yaml [hunk 5]
```diff
@@ -275,6 +275,10 @@ spec:
                     description: Console defines the installation output device.
                     pattern: ^(|tty[0-9]+|ttyS[0-9]+(,\d+([a-zA-Z0-9]+)?)?|ttyUSB[0-9]+(,\d+([a-zA-Z0-9]+))?|lp[0-9]+)$
                     type: string
+                  hwSettle:
```
[the remark this answers is not in the document]
Comment (sselvara1) [reply config/crd/bases/starlingx.windriver.com_hosts.yaml:278]: Re-generated CRD for update after adding the validation.

**Comment by sselvara1** [comment]
> Need to re-generate the chart to update the validation

Re-generated chart to update the validation.

**Comment by sselvara1** [comment]
> Apart from the CRDs, the code looks fine to me. Just checked with the dev prime of this feature, the update of this argument is time sensitive, needs to have a re-test in a standard system, and the change in the kernel should also be checked/verified as the validation of this change.

ok, will share this PR with dev prime of this feature to have a re-test in a standard system and also to check/verify the change in kernel.

**Review by yjian118 — APPROVED** [review]
Approve as tested in hw lab

**Review by yjian118 — CHANGES_REQUESTED** [review]
Need to rebase and re-build the tarball to address the merge conflict

File: go.mod (hunk 2 above)
Comment (ttakenak) [reply go.mod:94]: We don't need to put it back to "Wind-River"?

Comment (yjian118) [reply go.mod:94]: Yes, we do need to put it back, good catch, thanks!

**Review by yjian118 — CHANGES_REQUESTED** [review]
Need to update the go.mod

**Comment by sselvara1** [comment]
> We don't need to put it back gophercloud link to "Wind-Riiver" in go.mod?

yeah, gophercloud link is put back to "Wind-Riiver" in go.mod.

**Review by yjian118 — APPROVED** [review]

**Review by ttakenak — APPROVED** [review]

**Pull request merged**